coil-base/src/main/java/coil/drawable/CrossfadeDrawable.kt

```diff
@@ -24,14 +24,14 @@ import kotlin.math.roundToInt
  *
  * NOTE: The animation can only be executed once as the [start] drawable is dereferenced at the end of the transition.
  *
- * @param start The [Drawable] to crossfade from.
+ * @param start The [Drawable] to crossfade from. Will be mutated to prevent modifying alpha of original drawable
  * @param end The [Drawable] to crossfade to.
  * @param scale The scaling algorithm for [start] and [end].
  * @param durationMillis The duration of the crossfade animation.
  * @param fadeStart If false, the start drawable will not fade out while the end drawable fades in.
  */
 class CrossfadeDrawable @JvmOverloads constructor(
-    private var start: Drawable?,
+    start: Drawable?,
     private val end: Drawable?,
```

Comment:
We should probably mutate `end` as well since it could be loaded from resources.

Reply:
Yep, didn't think about that! Since we can provide something like error placeholder. Tested it in sample, same bug
https://github.com/coil-kt/coil/issues/571#issuecomment-722013475

Reply:
Fixed. Not sure if tests are good enough, they are testing drawable changes, but with a bit of a hack  - since percent of animation is tied to `SystemClock` and can't be influenced directly.